### PR TITLE
Preventing user from marking own expense as spam

### PR DESCRIPTION
**Pull request by samkevin1**
Resolve (https://github.com/opencollective/opencollective/issues/7159)

# Description

These changes will:
- Prevent the user from marking their own expense as spam
- Change the textarea placeholder in the "Mark as Spam" modal to a more fitting example
- Add a label explaining what is the textarea

# Screenshots

image

image

### CODE REVIEW

File: components/expenses/ConfirmProcessExpenseModal.tsx [hunk 1]
```diff
@@ -236,9 +242,19 @@ export default function ConfirmProcessExpenseModal({ type, onClose, expense }: C
...
       <ModalBody pt={2}>
-        <P mb={3} color="black.700" lineHeight="20px">
-          {intl.formatMessage(MessagesPerType[type].description)}
-        </P>
+        <Flex>
+          <AlertTriangleIcon size={30} color="orange" />
+          <P mb={3} color="black.700" lineHeight="20px" ml={10}>
+            {intl.formatMessage(MessagesPerType[type].description)}
+          </P>
+        </Flex>
```
Comment (Betree) [inline components/expenses/ConfirmProcessExpenseModal.tsx:250]: We don't have this pattern in our design system. If you want to highlight this message, I can recommend using a `<MessageBox type="warning" withIcon>` instead.

File: components/expenses/ConfirmProcessExpenseModal.tsx [hunk 2]
```diff
@@ -236,9 +242,19 @@ export default function ConfirmProcessExpenseModal({ type, onClose, expense }: C
...
+
+        {MessagesPerType[type].label && (
+          <P mb={2} color="black.700" fontWeight="500">
+            {intl.formatMessage(MessagesPerType[type].label)}
+          </P>
+        )}
+
         <RichTextEditor
           data-cy="confirm-action-text"
```
Comment (Betree) [inline components/expenses/ConfirmProcessExpenseModal.tsx:259]: For accessibilty, let's use proper labels and `htmlFor`

```suggestion
          <P as="label" htmlFor="expense-ban-reason" mb={2} color="black.700" fontWeight="500">
            {intl.formatMessage(MessagesPerType[type].label)}
          </P>
        )}

        <RichTextEditor
          id="expense-ban-reason"
          data-cy="confirm-action-text"
```

**Review by Betree — COMMENTED** [review]
That's a good start! You'll need to run lint, prettier, and `npm run build:langs` to get CI to pass.

I would also remove the "Are you sure?" below:

image

File: components/expenses/ConfirmProcessExpenseModal.tsx [hunk 3]
```diff
@@ -236,9 +242,19 @@ export default function ConfirmProcessExpenseModal({ type, onClose, expense }: C
...
+        <Flex>
+          <AlertTriangleIcon size={30} color="orange" />
+          <P mb={3} color="black.700" lineHeight="20px" ml={10}>
+            {intl.formatMessage(MessagesPerType[type].description)}
+          </P>
+        </Flex>
+
+        {MessagesPerType[type].label && (
+          <P mb={2} color="black.700" fontWeight="500">
+            {intl.formatMessage(MessagesPerType[type].label)}
```
Comment (Betree) [inline components/expenses/ConfirmProcessExpenseModal.tsx:254]: Let's also clarify that it's optional:

```suggestion
            {intl.formatMessage({ id: 'OptionalFieldLabel', defaultMessage: "{field} (optional)" }, {
              field: intl.formatMessage(MessagesPerType[type].label)
            )}
```

**Comment by samkevin1** [comment]
@Betree Hey! Made the changes you requested

File: components/expenses/ConfirmProcessExpenseModal.tsx [hunk 4]
```diff
@@ -14,7 +15,7 @@ import { toast } from '../ui/useToast';
 
 const messages = defineMessages({
   reasonPlaceholder: {
-    defaultMessage: 'e.g. Email Address is wrong',
+    defaultMessage: 'e.g. Same expense was submitted twice.',
```
Comment (Betree) [inline components/expenses/ConfirmProcessExpenseModal.tsx:18]: We need a stronger reason here

```suggestion
    defaultMessage: "e.g., We never worked with this person.",
```

File: components/expenses/ExpenseMoreActionsButton.js [hunk 5]
```diff
@@ -118,6 +122,17 @@ const ExpenseMoreActionsButton = ({
                 buttonStyle="dangerSecondary"
                 data-cy="spam-button"
                 onClick={async () => {
+                  const isSubmitter = expense.createdByAccount.legacyId === LoggedInUser?.CollectiveId;
+
+                  if (isSubmitter) {
+                    toast({
+                      variant: 'error',
+                      message: "You can't mark your own expenses as spam",
```
Comment (Betree) [inline components/expenses/ExpenseMoreActionsButton.js:130]: This needs internationalization

**Comment by samkevin1** [comment]
@Betree Hey! Made the changes

File: components/expenses/ConfirmProcessExpenseModal.tsx [hunk 6]
```diff
@@ -14,7 +15,7 @@ import { toast } from '../ui/useToast';
 
 const messages = defineMessages({
   reasonPlaceholder: {
-    defaultMessage: 'e.g. Email Address is wrong',
+    defaultMessage: 'e.g. We never worked with this person.',
```
Comment (Betree) [inline components/expenses/ConfirmProcessExpenseModal.tsx:18]: We use the American version for this.

```suggestion
    defaultMessage: 'e.g., We never worked with this person.',
```

File: components/expenses/ExpenseMoreActionsButton.js [hunk 7]
```diff
@@ -86,11 +90,18 @@ const ExpenseMoreActionsButton = ({
...
   const showDeleteConfirmMoreActions = isOpen => {
     setDeleteConfirm(isOpen);
     onModalToggle?.(isOpen);
   };
 
+  const spamNotAllowedMessage = defineMessage({
+    id: 'expense.spam.notAllowed',
+    defaultMessage: "You can't mark your own expenses as spam",
+  });
+
```
Comment (Betree) [inline components/expenses/ExpenseMoreActionsButton.js:104]: No need to do that, you can directly pass the object to `intl.formatMessage`

File: components/expenses/ProcessExpenseButtons.js [hunk 8]
```diff
@@ -149,6 +151,11 @@ const ProcessExpenseButtons = ({
   const intl = useIntl();
   const { toast } = useToast();
 
+  const spamNotAllowedMessage = defineMessage({
+    id: 'expense.spam.notAllowed',
+    defaultMessage: "You can't mark your own expenses as spam",
+  });
+
```
Comment (Betree) [inline components/expenses/ProcessExpenseButtons.js:158]: Same here

**Review by Betree — COMMENTED** [review]
Added some additional feedback. Let me know if you want me to take care of the final cleanup to get the PR ready.

**Comment by samkevin1** [comment]
@Betree Hey! Made the changes, was that the cleanup you were referring to?

**Comment by samkevin1** [comment]
@Betree Hey! Any feedback on this one?

File: components/expenses/ProcessExpenseButtons.js [hunk 9]
```diff
@@ -401,4 +421,4 @@ ProcessExpenseButtons.defaultProps = {
   isViewingExpenseInHostContext: false,
 };
 
-export default ProcessExpenseButtons;
+export default withUser(ProcessExpenseButtons);
```
[the remark this answers is not in the document]
Comment (samkevin1) [reply components/expenses/ProcessExpenseButtons.js]: Updated this @Betree. Could you take another look? Also not sure why the ci is failing

[the remark this answers is not in the document]
Comment (samkevin1) [reply components/expenses/ProcessExpenseButtons.js]: Any update on this @Betree one?

**Review by Betree — APPROVED** [review]
Working great, thank you @samkevin1!

**Pull request merged**